### PR TITLE
Store and reuse client certificate task

### PR DESCRIPTION
**Pull request by Kahbazi**

### CODE REVIEW

File: src/Middleware/HttpOverrides/src/CertificateForwardingFeature.cs [hunk 1]
```diff
@@ -10,10 +10,12 @@ namespace Microsoft.AspNetCore.HttpOverrides;
 
 internal sealed class CertificateForwardingFeature : ITlsConnectionFeature
 {
+    private static readonly Task<X509Certificate2?> _nullCertificate = Task.FromResult<X509Certificate2?>(null);
```
Comment (gfoidl) [inline src/Middleware/HttpOverrides/src/CertificateForwardingFeature.cs:13]: This isn't needed (IMO).
* it's only use in the catch block, so in a exceptional case
* `Task.FromResult` uses a (internally) cached task (for default-values, which is `null` for reference types)

**Comment by adityamandaleeka** [comment]
@wtgodbe Can you please review this one?

File: src/Middleware/HttpOverrides/src/CertificateForwardingFeature.cs (hunk 1 above)
Comment (wtgodbe) [reply src/Middleware/HttpOverrides/src/CertificateForwardingFeature.cs:13]: Agree with this

**Review by wtgodbe — APPROVED** [review]
LGTM, but I'd get rid of `_nullCertificate`

**Pull request merged**